vunit/vhdl_parser.py

```diff
@@ -85,9 +85,16 @@ def parse(cls, code):
         )
 
     _component_re = re.compile(
-        r"[a-zA-Z]\w*\s*\:\s*(?:component)?\s*(?:(?:[a-zA-Z]\w*)\.)?([a-zA-Z]\w*)\s*"
-        r"(?:generic|port) map\s*\([\s\w\=\>\,\.\)\(\+\-\'\"]*\);",
-        re.IGNORECASE,
+        r"""
+        [a-zA-Z]\w*               # Label
+        \s*\:\s*                  # Semicolon
+        (?:component)?\s*         # Optional component keyword
+        (?:(?:[a-zA-Z]\w*)\.)?    # Optional library name
+        ([a-zA-Z]\w*)\s*          # Capture component name
+        (?:generic|port)\s+map\s* # Generic/port map
+        \(.*?\)\s*;               # Open and closing brackets
```

Comment:
Rather than allowing anything between the opening and closing brackets I would suggest extending the previously more restrictive expression with what is needed to pass the new test case:

\([\s\w\=\>\,\.\)\(\+\-\'\"\*/]*\)

I cannot recall if the more restrictive expression is there for a reason that is not made explicit in the test cases. For that reason I prefer a more defensive modification.